### PR TITLE
evm: remove wrong error at GetPrecompiledContractMap

### PR DESCRIPTION
**Pull request by yoomee1313**
## Proposed changes

- If the addr is an EOA, it is returned at `default` case.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: blockchain/vm/evm.go [hunk 1]
```diff
@@ -572,8 +572,6 @@ func (evm *EVM) GetPrecompiledContractMap(addr common.Address) map[common.Addres
 	case evm.chainRules.IsIstanbul:
 		return PrecompiledContractsIstanbulCompatible
 	default:
-		// It should not happen.
-		logger.Error("VmVersion is not zero, but it uses PrecompiledContractsByzantiumCompatible")
```
Comment (jeongkyun-oh) [inline blockchain/vm/evm.go:576]: I couldn't see when an EOA addresss can be a param of this method `GetPrecompiledContractMap`. Is there any case?

Comment (yoomee1313) [reply blockchain/vm/evm.go:576]: @jeongkyun-oh EOA address can be inserted as an `addr` when caller is `isProgramAccount`. 
https://github.com/klaytn/klaytn/blob/d7a817488a2aa86a6343e127e1f970a460ebd9c9/blockchain/vm/evm.go#L55-L61

**Review by jeongkyun-oh — APPROVED** [review]

**Review by JayChoi1736 — APPROVED** [review]

**Comment by blukat29** [comment]
Late to the party, but I just reproduced a benign case.

The log indeed prints before Istanbul hardfork, by a transaction from an EOA to an SCA fallback function.
```js
> var A = personal.listAccounts[0]
> var txid = klay.sendTransaction({ from: A, data: "0x6080604052348015600f57600080fd5b50604580601d6000396000f3fe608060405236600a57005b600080fdfea26469706673582212204422478d51beb7ab21b569adef4dee784042e112dde8b0dfc620f76a34a4f5aa64736f6c63430008070033" })
> var sca = klay.getTransactionReceipt(txid).contractAddress
> klay.sendTransaction({ from: A, to: sca })
```

**Comment by kjeom** [comment]
I think we need to change the comment at [no more internal function](https://github.com/klaytn/klaytn/pull/1723/files#diff-1ac4ee7ce2a68b1012577e8183ae43f6ed4d82584240ee85f6d67c4754ad250cR561)

**Comment by yoomee1313** [comment]
@blukat29 The example you showed needs to print the log(if this PR is not applied) because i think it calls `evm.Call` -> `isProgramAccount`. btw, EOA does not have VmVersion field, so regardless of the hard fork version, VmVersion is not used.

**Review by kjeom — APPROVED** [review]

**Review by jeongkyun-oh — APPROVED** [review]

**Pull request merged**